proposals/target-typed-new.md

```diff
@@ -0,0 +1,99 @@
+
+# Target-typed `new` expressions
+
+* [x] Proposed
+* [x] [Prototype](https://github.com/alrz/roslyn/tree/features/target-typed-new)
+* [ ] Implementation
+* [ ] Specification
+
+## Summary
+[summary]: #summary
+
+Do not require type specification for constructors when the type is known. 
+
+## Motivation
+[motivation]: #motivation
+
+Allow field initialization without duplicating the type.
+```cs
+Dictionary<string, List<int>> field = new() {
+    { "item1", new() { 1, 2, 3 } }
+};
+```
+Allow omitting the type when it can be inferred from usage.
+```cs
+XmlReader.Create(reader, new() { IgnoreWhitespace = true });
+```
+
+## Detailed design
+[design]: #detailed-design
+
+The *object_creation_expression* syntax would be modified to make the *type* optional when parentheses are present. This is required to address the ambiguity with *anonymous_object_creation_expression*.
+```antlr
+object_creation_expression
+    : 'new' type? '(' argument_list? ')' object_or_collection_initializer?
+    | 'new' type object_or_collection_initializer
+    ;
+```
+The target-type is resolved through a conversion that only exists if and only if there exists a single best accessible constructor with regard to provided arguments. Target of the initialization is determined after the conversion succeeded.
+
+The type of the expression would be inferred from the target-type which would be required to be one of the following:
+
+- **Any struct type**
+- **Any reference type**
+- **Any type parameter** with a constructor constraint
+
+with the following exceptions:
+
+- **Enum types:** not all enum types contain the constant zero, so it should be desirable to use the explicit enum member.
+- **Delegate types:** if the type is inferrable, an anonymous function can already be used.
+- **Interface types:** this is a niche feature and it should be preferable to explicitly mention the type.
+- **Tuple types:** tuple literals should be used for such expressions.
+- **Array types:** arrays need a special syntax to provide the length.
+
+All the other types that are not permitted in the *object_creation_expression* are excluded as well, for instance, pointer types.
+
+The default constructor for a value type (unless appeared explicitly in metadata) is ought to be excluded, which would have the effect of excluding the primitive types and the default constructor for most value types. If you wanted to use the default value of such types you could write `default`.
```

Comment:
>is ought [](start = 82, length = 8)

"is" or "ought to be"? ... "would have" or "has" the effect? Please take a position one way or the other (though whatever position you take you can leave it as an open issue that we might reconsider).

Reply:
I was just quoting your words, I think that's why it came out wired.